### DISCUSSION TITLE
defaultInstallRoot setting does not take effect

### DISCUSSION THREAD

**Discussion by dong-lufei** [microsoft/winget-cli#2850] [Q&A] [answered]
Regardless of the exe or msi format, the path specified by defaultInstallRoot does not take effect after installation

```
# 13697 @ donglufei in E:\test [18:35:19]
$ cat C:\Users\13697\AppData\Local\Packages\Microsoft.DesktopAppInstaller_8wekyb3d8bbwe\LocalState\settings.json
{
  "$schema": "https://aka.ms/winget-settings.schema.json",

  "installBehavior": {
    // 安装软件包的默认位置。这可以被--location参数覆盖。此设置仅在包清单包含 时使用InstallLocationRequired，并且通过将包 ID 附加到根来获取实际位置
    "defaultInstallRoot": "D:/wingetApps"
  },
  "visual": {
    //指定显示的进度条的颜色
    "progressBar": "rainbow" // 默认重音accent，复古的retro、彩虹rainbow
  }
}

# 13697 @ donglufei in E:\test [18:37:24]
```
winget install appX （appX is in msi format, using -l parameter also fails, using -i succeeds）

winget install appX （appX is in exe format, using -l parameter also succeeds, using -i succeeds）

**Comment by Trenly** [comment, marked as answer]
I believe this is because the setting only takes effect "If the package requires one". As noted in the settings schema, this setting is only used when a package manifest includes `InstallLocationRequired`
https://github.com/microsoft/winget-cli/blob/fb4bf346ad59a7222ebe3c82cc5573e8494c3eef/doc/Settings.md?plain=1#L140
